### ISSUE TITLE
ARIA2: Conflict with other ARIA-related specs

### ISSUE THREAD

**Issue by momdo** [w3c/wcag#145]
<a href="https://www.w3.org/WAI/GL/2016/WD-WCAG20-TECHS-20160105/ARIA2.html">Current ARIA2</a> in WCAG20-TECHS showcases the use of the `aria-required` property to indicate requirement of field input.
However, HTML5 has already defined the <a href="https://www.w3.org/TR/html5/forms.html#the-required-attribute"><code>required</code></a> attribute, by which the same objective should be achieved.

<a href="https://www.w3.org/TR/wai-aria-1.1/#host_general_conflict">According to WAI-ARIA 1.1</a>, authors should use features provided by HTML, in stead of ones from WAI-ARIA, as long as "a feature in the host language with identical role semantics and values is available, and the author has no compelling reason to avoid using the host language feature". Considering that the `required` HTML attribute and the `aria-required` ARIA property are considered to be identical [1] \(and is also is recommended to prefer the HTML one to the ARIA one where HTML ones are applicable [2]), examples in ARIA2 conflict with other related specifications.

The idea itself to mandate the indication of input requirements is favourable from the view point of accessibility for users (and, interestingly, for machines, too). Also for the philosophy of the idea, modification on this section is required to make the guideline conformant with existing specifications.

[1] https://www.w3.org/TR/html-aria/#attr-required
[2] https://www.w3.org/TR/html-aria/#h-note2

**Comment by jnurthen** [comment]
Authors want to have a consistent experience across user agents. The HTML required attribute changes user agent behavior. Using aria-required instead allows developers to create a consistent experience across platforms while still maintaining a11y support.

**Comment by awkawk** [comment]
we could and should clarify that HTML5 includes the required attribute and that authors should consider using that first and then if that is not able to be used to go the aria-required route instead.  How does that sound?

**Comment by joshueoconnor** [comment]
Theres another reason to +1 @jnurthen suggestion because in the real world support for aria-required is better, at least from a legacy perspective.

**Comment by jnurthen** [comment]
@awkawk I disagree. These are ARIA techniques. I'm not sure we should document that there are other HTML techniques in each and every one. Pointing to an HTML required technique should be left to the related techniques section (note there is no current technique about using @required)

**Comment by awkawk** [comment]
Preparing to mark as deferred due to inactivity and group resource constraints (https://www.w3.org/WAI/GL/wiki/Handling_Issues).
